packages/container/lib/container.js

```diff
@@ -559,6 +559,7 @@ function factoryInjectionsFor(container, fullName) {
 
 // TODO - remove when Ember reaches v3.0.0
 function injectDeprecatedContainer(object, container) {
+  if ('container' in object) { return; }
```

Comment:
This isn't quite right, but it may not matter at this point. In the case above (in `instantiate`) where we are creating an instance on a factory that does not have `.extend` we have to ensure that the `.container` property on the instance (which may be the `container._fakeContainerToInject` is replaced by a getter that issues a deprecation. 

Now, considering that this has been deprecated for quite some time (since ember 2.3), it's might be fine to drop support for the deprecation in this odd edge case.

I'll work up a failing test case so you can see what I mean.